lib/respec/respec-writer.ts

```diff
@@ -19,15 +21,25 @@ export function writeReSpec(
   }
   const outputStream = fileSystem.createWriteStream(
     path.join(directory, name + ".html"));
-  outputStream.write("<!DOCTYPE html>\n");
-  outputStream.write("<html lang=\"cs\">");
-  writeHeader(model, outputStream);
-  writeBody(model, outputStream);
-  outputStream.write("\n</html>\n");
+  constructReSpec(model, outputStream);
   outputStream.close();
 }
 
-function writeHeader(model: ReSpec, stream: WriteStream) {
+export function getReSpec(model: ReSpec): string {
+  const stream = new StringWriteStream();
+  constructReSpec(model, stream);
+  return stream.getContent();
+}
+
+function constructReSpec(model: ReSpec, stream: WritableObject) {
+  stream.write("<!DOCTYPE html>\n");
+  stream.write("<html lang=\"cs\">");
+  writeHeader(model, stream);
+  writeBody(model, stream);
+  stream.write("\n</html>\n");
+}
+
+function writeHeader(model: ReSpec, stream: WritableObject) {
```

Comment:
I may overlooked that in the BiShed writer, but you have parameter named "stream" with type "WritableObject" , those two do not match together.